### ISSUE TITLE
Large message channel and ask

### ISSUE THREAD

**Issue by johanandren** [akka/akka#29205]
When using the large message channel a user needs control over where the messages are sent (to maintain message ordering) but if adding ask to the mix, the response will go through the PromiseActorRef which will always live under `/tmp`.

A workaround for users would be to avoid using ask and introducing their own child to keep track of the timeout etc. and mark children for large message channel using the config.

Would be nice if they didn't have to duplicate but could use ask though, is there some way we could improve on this?

**Comment by patriknw** [comment]
The path of the PromiseActorRef is currently:
```
  private val tempNode = rootPath / "temp"

  override def tempPath(): ActorPath = tempNode / Helpers.base64(tempNumber.getAndIncrement())
```

One idea is that we would add a prefix to the tempNumber part. Then `/temp/<prefix>*` can be defined as a wildcard for the large messages.

What to base the prefix on?
Could be the target path or the message class. I think we have both these in some shape when creating the PromiseActorRef.

(Could also be pretty useful for debugging, to see which ask a "temp" ref belongs to.)

**Comment by johanandren** [comment]
Started with target encoded in name, tricky figuring out a reasonable and readable escaped name though.

Tried URL-encoding first but that just becomes letter soup. 
A few ideas I played with, not entirely happy with any of them:
```
akka://AskSpec/temp/user|silent$a
akka://AskSpec/temp/[user,silent]$a
akka://AskSpec/temp/user_silent$a
```

Also, since we want to be able to match on prefix we cannot include target system in there (or at least no in the natural place in the beginning of the prefix), which makes it less useful for debugging asks in a cluster.

**Comment by patriknw** [comment]
It doesn't have to be unique since we should still append the tempNumber at the end.
Maybe enough to use `name` of the target path (only last element)? They are in control of the naming of those actors I suppose so can be given a unique enough name for the wildcard.

The reason I was thinking of including the message class was if not all replies are large, but that is probably less important.

**Comment by johanandren** [comment]
The message class could be better because it is already closed over, while adding the target (or sender) will add some overhead to all asks (now the target is captured and lazily turned into a string only if the ask times out).

**Comment by patriknw** [comment]
This has been completed and included in Akka 2.6.7 and 2.6.8, maybe even 2.6.6
